src/prefect/server/api/deployments.py

```diff
@@ -650,6 +652,9 @@ async def create_flow_run_from_deployment(
                     detail="Invalid schema: Unable to validate schema with circular references.",
                 )
 
+        if PREFECT_EXPERIMENTAL_ENABLE_FLOW_RUN_INFRA_OVERRIDES.value():
+            validate_job_variables_for_flow_run(flow_run, deployment)
```

Comment:
It looks like you can create a flow run with job variables if the flag is turned off, but they won't be validated. Would it make sense to strip off `job_variables` if the flag is off, or will that be handled downstream?

Reply:
The way I've been thinking about it is that `job_variables` can be included in fields/responses but nothing will happen with them unless the flag is on (validation won't happen, and workers won't use them). Effectively the field is ignored until the flag is enabled.

Having  workers use them is coming in a follow up PR!